include/envoy/buffer/buffer.h

```diff
@@ -18,6 +18,29 @@ struct RawSlice {
   size_t len_ = 0;
 };
 
+/**
+ * A wrapper class to facilitate passing in externally owned data to a buffer via addBufferFragment.
+ * When the buffer no longer needs the data passed in through a fragment, it calls done() on it.
+ */
+class BufferFragment {
+public:
+  virtual ~BufferFragment(){};
+  /**
+   * @return a pointer to the referenced data.
+   */
+  virtual const void* data() PURE;
+
+  /**
+   * @return the size of the referenced data.
+   */
+  virtual size_t size() PURE;
```

Comment:
This can probably safely be marked const here (no implementation should need a non-const size()): `virtual size_t size() const PURE`
 
The comment should have the return type like: `@return size_t the size of ...`